_source/_posts/2023-03-22-stepup-okta.md

```diff
@@ -0,0 +1,247 @@
+    "updated_at": 1490198843,
+    "zoneinfo": "America/Los_Angeles"
+}
+``` 
+``` 
+curl -v -X GET \
+-H "Accept: application/json" \
+-H "Content-Type: application/json" \
+-H "Authorization: SSWS ${api_token}" \
+"https://${yourOktaDomain}/api/v1/users?search=profile.login+eq+%22login%40example.com%22"
+``` 
+
+* Find the Factor ID for the enrolled user
+
+The Factor ID is available from the [factor management API](https://developer.okta.com/docs/reference/api/factors/#list-enrolled-factors)
+
+``` 
+curl -v -X GET \
+-H "Accept: application/json" \
+-H "Content-Type: application/json" \
+-H "Authorization: SSWS ${api_token}" \
+"https://${yourOktaDomain}/api/v1/users/00u15s1KDETTQMQYABRL/factors"
+``` 
+
+* Example: Issue an SMS Factor challenge
+
+``` 
+curl -v -X POST \
+-H "Accept: application/json" \
+-H "Content-Type: application/json" \
+-H "Authorization: SSWS ${api_token}" \
+-d '{
+}' "https://${yourOktaDomain}/api/v1/users/00u15s1KDETTQMQYABRL/factors/smsszf1YNUtGWTx4j0g3/verify"
+```
+
+* Example: Verify an SMS Factor challenge
+
+```
+curl -v -X POST \
+-H "Accept: application/json" \
+-H "Content-Type: application/json" \
+-H "Authorization: SSWS ${api_token}" \
+-d '{
+  "passCode": "123456"
+}' "https://${yourOktaDomain}/api/v1/users/00u15s1KDETTQMQYABRL/factors/smsszf1YNUtGWTx4j0g3/verify"
+```
+
+
+
+## Okta Access Gateway
+
+**Scenario**
+
+* Legacy web application supporting header based authentication 
+
+* The application is integrated with WIC leveraging [Okta Access Gateway(OAG)](https://www.okta.com/products/access-gateway/) Web Access Management (WAM) solution
+
+**Solution**
+
+* OAG can enforce step-up authentication for sensitive resources by cleverly crafting the [advanced policies](https://help.okta.com/oag/en-us/Content/Topics/Access-Gateway/advanced-oag-policy-development.htm)
+
+
+## Transactional MFA
+
+Transactional MFA is a close cousin of step-up authentication. In some step-up cases, transactional MFA can be used instead.
+
+**Scenario**
+
+* User authenticates to the application. 
+
+* At some point the user tries to access a sensitive resource.
+
+* User receives a push notification in her mobile to approve the transaction. 
+
+* Once approved, the access is granted to the resource.
+
+**Solution**
+
+Okta supports the above transactional MFA scenario using [OIDC Client-Initiated Backchannel Authentication Flow (CIBA)](https://developer.okta.com/docs/guides/configure-ciba/main/)
+
+We will look at CIBA based scenario in more detail in a later article.
+
+
+## In summary
+
+Whether you application is using modern protocols such as OpenID Connect or built from legacy technologies, Okta's Workforce Identity Cloud can help you prompt users to step-up their authentication only when necessary for accessing sensitive resources. 
+
+Use your [Okta Developer Account](https://developer.okta.com/signup/) to test different step-up authentication flows and determine which is best for your application!
```

Comment:
We should add CTA here to follow OktaDev on YouTube, Twitter, and perhaps some other posts that may be of interest.